src/librustc/hir/mod.rs

```diff
@@ -1497,8 +1468,16 @@ pub enum Item_ {
     ///
     /// e.g. `extern crate foo` or `extern crate foo_bar as foo`
     ItemExternCrate(Option<Name>),
-    /// A `use` or `pub use` item
-    ItemUse(P<ViewPath>),
+
+    /// `use foo::bar::baz as quux;`
+    ///
+    /// or just
+    ///
+    /// `use foo::bar::baz;` (with `as baz` implicitly on the right)
+    ItemUse(P<Path>),
+
+    /// `use foo::bar::*;`
+    ItemUseGlob(P<Path>),
```

Comment:
It'd be simpler to use `ItemUse(P<Path>, bool /* is glob*/)`/`ItemUse(P<Path>, UseKind)`, this distinction is very rarely relevant in HIR and I'd prefer not to see it in every exhaustive match.